kogito-codegen-modules/kogito-codegen-processes/src/main/java/org/kie/kogito/codegen/process/TagResourceGenerator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.kogito.codegen.process;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+import org.jbpm.ruleflow.core.Metadata;
+import org.kie.kogito.internal.process.runtime.KogitoWorkflowProcess;
+
+import com.github.javaparser.ast.CompilationUnit;
+import com.github.javaparser.ast.NodeList;
+import com.github.javaparser.ast.body.ClassOrInterfaceDeclaration;
+import com.github.javaparser.ast.expr.MemberValuePair;
+import com.github.javaparser.ast.expr.Name;
+import com.github.javaparser.ast.expr.NormalAnnotationExpr;
+import com.github.javaparser.ast.expr.StringLiteralExpr;
+
+/**
+ * Adds {@code org.eclipse.microprofile.openapi.annotations.tags.Tag} annotations to a {@link CompilationUnit}
+ */
+final class TagResourceGenerator {
+
+    private TagResourceGenerator() {
+    }
+
+    /**
+     * Adds {@code org.eclipse.microprofile.openapi.annotations.tags.Tag} annotations to the specified {@link CompilationUnit}
+     *
+     * @param compilationUnit the compilation unit to add the {@code org.eclipse.microprofile.openapi.annotations.tags.Tag} annotations to
+     * @param process the {@link KogitoWorkflowProcess} to get the tags from
+     */
+    static void addTags(CompilationUnit compilationUnit, KogitoWorkflowProcess process) {
+        Map<String, Object> metadata = process.getMetaData();
+        compilationUnit.findAll(ClassOrInterfaceDeclaration.class)
+                .forEach(cls -> addTags(process, (Collection<String>) metadata.getOrDefault(Metadata.TAGS, List.of()), (String) metadata.get(Metadata.DESCRIPTION), cls));
```

Comment:
Wouldn't be better to get metadata and description only once rather than on each iteration?

```suggestion
        @SuppressWarnings("unchecked") 
        Collection<String> tags = (Collection<String>) metadata.getOrDefault(Metadata.TAGS, List.of());
        String description = (String) metadata.get(Metadata.DESCRIPTION);
        compilationUnit.findAll(ClassOrInterfaceDeclaration.class)
                .forEach(cls -> addTags(process, tags, description, cls));
```